chaos_genius/alerts/anomaly_alerts.py

```diff
@@ -171,56 +237,49 @@ def _find_point(self, point, prev_data):
-
-            if intended_point is None:
-                # previous point wasn't found
-                point["percentage_change"] = "–"
-            elif point["y"] == 0 and intended_point["y"] == point["y"]:
-                # previous data was same as current
-                point["percentage_change"] = "–"
-            elif intended_point["y"] == 0:
-                # previous point was 0
-                sign_ = "+" if point["y"] > 0 else "-"
-                point["percentage_change"] = sign_ + "inf"
-            else:
-                point["percentage_change"] = find_percentage_change(
-                    point["y"], intended_point["y"]
-                )
+    def _save_nl_message(self, anomaly_data: List[dict]):
+        """Constructs and saves change message for every point."""
+        
+        kpi = Kpi.get_by_id(self.kpi_id)
+        if kpi is None:
+            for point in anomaly_data:
+                point["nl_message"] = "KPI does not exist"
+            return
 
-            point["nl_message"] = change_message_from_percent(
-                point["percentage_change"]
-            )
+        time_series_freq = kpi.anomaly_params.get("frequency")
+        if time_series_freq is None or time_series_freq not in ("d", "D", "Daily", "daily", "h", "H", "hourly", "Hourly"):
+            for point in anomaly_data:
+                point["nl_message"] = "Time series frequency does not exist"
+            return
 
-    def _save_nl_message_hourly_freq(self, anomaly_data: List[dict], kpi: Kpi):
-        """Saves change message for every point, for a hourly frequency KPI."""
-        data = dict()
         time_diff = datetime.timedelta(days=1, hours=0, minutes=0)
 
-        # TODO: fix circular import
-        from chaos_genius.controllers.digest_controller import get_previous_data
+        last_anomaly_timestamp: Optional[datetime.datetime] = self.alert_info[
+            "last_anomaly_timestamp"
+        ]
+
+        if last_anomaly_timestamp is not None:
+            # when last_anomaly_timestamp is available
+            #   get data after last_anomaly_timestamp
+            start_timestamp = last_anomaly_timestamp - time_diff
+            include_start_timestamp = False
+        else:
+            # when last_anomaly_timestamp is not available
+            #   get data of the last timestamp in anomaly table
+            start_timestamp = self.latest_anomaly_timestamp - time_diff
+            include_start_timestamp = True
+
+        prev_day_data = get_anomaly_data(
+            [self.kpi_id],
+            anomaly_types=["subdim", "overall"],
+            start_timestamp=start_timestamp,
+            include_start_timestamp=include_start_timestamp,
+            # only get anomaly data till latest timestamp
+            #   (ignore newer data added after alert started)
+            end_timestamp=self.latest_anomaly_timestamp - time_diff,
+            include_end_timestamp=True,
+        )
```

Comment:
We're trying to reduce duplicate code, not increase it :)

You can add a parameter to `_get_anomalies` for the time offset.